### PR TITLE
Embed Plugin (inline)

### PR DESCRIPTION
**Pull request by danarish4**

### CODE REVIEW

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx [hunk 1]
```diff
@@ -27,17 +31,33 @@ class LinkPreviewViewer extends Component {
 
   componentDidMount() {
     validate(pluginLinkPreviewSchema, this.props.componentData);
-    this.setState({ imageHeight: this.image.offsetHeight });
+    this.setState({ imageHeight: this.image?.offsetHeight });
+    const { componentData } = this.props;
+    const { html } = componentData;
+    const iframeDocument = document.getElementById('myiframe')?.contentWindow?.document;
+    if (iframeDocument) {
```
Comment (oded-soffrin) [inline packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: similar code here: https://github.com/wix-incubator/rich-content/blob/master/packages/plugin-html/web/src/IframeHtml.jsx#L22

Comment (oded-soffrin) [reply packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: if we can get to the state where all iframe logic is in HTML plugin that would be ideal
also. -getElementById('myiframe') won't work if we have many iframes

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx [hunk 2]
```diff
@@ -3,10 +3,14 @@ import PropTypes from 'prop-types';
 import { isEqual } from 'lodash';
 import { mergeStyles, validate, pluginLinkPreviewSchema } from 'wix-rich-content-common';
 import styles from '../statics/styles/link-preview.scss';
+import HtmlComponent from 'wix-rich-content-plugin-html/dist/lib/HtmlComponent';
 
 class LinkPreviewViewer extends Component {
   static propTypes = {
     componentData: PropTypes.object.isRequired,
+    settings: PropTypes.shape({
+      disableEmbed: PropTypes.bool,
```
Comment (oded-soffrin) [inline packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: suggestion - 'enableEmbed' instead disable, with default value (true / false ? ask product) 
IMO default value should be 'false' (no embed)

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx [hunk 3]
```diff
@@ -27,17 +31,25 @@ class LinkPreviewViewer extends Component {
...
   componentDidMount() {
     validate(pluginLinkPreviewSchema, this.props.componentData);
-    this.setState({ imageHeight: this.image.offsetHeight });
+    this.setState({ imageHeight: this.image?.offsetHeight });
   }
 
   getUrlForDisplay = url => url.replace(/^https?:\/\//, '');
 
+  handleIframeLoad = () => {
+    this.iframe.style.height = this.iframe.contentWindow.document.body.scrollHeight + 'px';
```
Comment (oded-soffrin) [inline packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: who is this.iframe this line is refering to?

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx [hunk 4]
```diff
@@ -53,26 +65,44 @@ class LinkPreviewViewer extends Component {
...
-      </figure>
-    );
+    if (settings.enableEmbed && html) {
+      const htmlCompProps = {
+        componentData: {
+          ...componentData,
+          srcType: 'html',
+          src: unescape(html),
+          config: {
+            height: this.iframe?.style.height,
```
Comment (oded-soffrin) [inline packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: same here (this.iframe)

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx (hunk 3 above)
Comment (danarish4) [reply packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: deleted

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx (hunk 4 above)
Comment (danarish4) [reply packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: deleted

File: packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx [hunk 5]
```diff
@@ -3,10 +3,14 @@ import PropTypes from 'prop-types';
...
 import { mergeStyles, validate, pluginLinkPreviewSchema } from 'wix-rich-content-common';
 import styles from '../statics/styles/link-preview.scss';
+import HtmlComponent from 'wix-rich-content-plugin-html/dist/lib/HtmlComponent';
 
 class LinkPreviewViewer extends Component {
   static propTypes = {
     componentData: PropTypes.object.isRequired,
+    settings: PropTypes.shape({
+      enableEmbed: PropTypes.bool,
+    }).isRequired,
```
Comment (oded-soffrin) [inline packages/plugin-link-preview/web/src/LinkPreviewViewer.jsx]: not required right? true by default

File: e2e/cypress/support/commands.js [hunk 6]
```diff
@@ -472,6 +472,15 @@ Cypress.Commands.add('waitForHtmlToLoad', () => {
     .wait(4000);
 });
 
+Cypress.Commands.add('convertLinkToPreview', url => {
```
Comment (oded-soffrin) [inline e2e/cypress/support/commands.js]: naming: insertLinkAndEnter ?

File: examples/storybook/stories/Plugins/LinkPreview.js [hunk 7]
```diff
@@ -32,7 +44,11 @@ export default () => {
...
         </RichContentEditorBox>
         <RichContentViewerBox preset="blog-preset">
-          <RichContentViewer initialState={fixtrue} typeMappers={typeMappers} />
+          <RichContentViewer
+            initialState={fixtrue}
+            typeMappers={typeMappers}
+            config={viewerConfig}
+          />
         </RichContentViewerBox>
       </Section>
```
Comment (oded-soffrin) [inline examples/storybook/stories/Plugins/LinkPreview.js]: Please insert some text (can be in the fixture itself):
"enter link to see link preview" ... "enter instagram link to see embed" ... NOTE - Mock only (use @wix/rich-content to consume link metadata fetcher service)

**Pull request merged**